internal/controller/warehouses/images.go

```diff
@@ -129,11 +129,20 @@ func imageDiscoveryLogFields(sub kargoapi.ImageSubscription) []any {
 		"platformConstrained", sub.Platform != "",
 	}
 	switch sub.ImageSelectionStrategy {
-	case kargoapi.ImageSelectionStrategySemVer, kargoapi.ImageSelectionStrategyDigest:
-		f = append(
-			f,
-			"semverConstraint", sub.SemverConstraint,
-		)
+	case kargoapi.ImageSelectionStrategySemVer:
+		if sub.Constraint != "" {
+			f = append(
+				f,
+				"constraint", sub.Constraint)
+		} else if sub.SemverConstraint != "" {
```

Comment:
This change would preserve the original behavior where lack of a constraint is still explicitly recorded, which I think is a beneficial detail.

```suggestion
		} else {
```